server/routes/viewModels/taskLists/caTasks.js

```diff
@@ -22,30 +22,41 @@ const eligibilityTask = namedTask('eligibilityTask')
 const eligibilitySummaryTask = namedTask('eligibilitySummaryTask')
 
 module.exports = {
-  getTasksForBlocked: (errorCode) => tasklist({}, [[eligibilityTask], [informOffenderTask], [caBlocked(errorCode)]]),
-
-  getCaTasksEligibility: ({ decisions, tasks, allowedTransition }) => {
+  getCaTasksEligibility: ({ decisions, tasks, allowedTransition, errorCode }) => {
     const { optedOut, eligible, bassReferralNeeded, addressUnsuitable } = decisions
-    const { eligibility, optOut } = tasks
+    const { eligibility, optOut, bassRequest } = tasks
 
     const eligibilityDone = eligibility === 'DONE'
     const optOutUnstarted = optOut === 'UNSTARTED'
     const optOutRefused = optOut === 'DONE' && !optedOut
 
+    // completed only when both curfewAddress & bassRequest are DONE (once one is done all sections curfew/CAS/opt out are done) and not opted out
+    const isCurfewAddressCompleted = optOutRefused && tasks.curfewAddress === 'DONE' && bassRequest === 'DONE'
```

Comment:
mmh.. does this work? - isn't BASS request only DONE if it's a BASS? 

is the logic to determine whether the case should be handed over already present - isn't it when either?:
* eligibleForBASSReview: `optOutRefused && bassReferralNeeded && allowedTransition !== 'caToDmRefusal']`
* eligibleForAddressReview: `optOutRefused && !bassReferralNeeded && allowedTransition !== 'caToDmRefusal'`

so:
```ts
const eligibleForHandover = eligibleForBASSReview || eligibleForAddressReview

[submitBassReview, eligibleForBassReview && !errorCode],
[submitAddressReview, eligibleForAddressReview && !errorCode],
[caBlocked, eligibleForHandover && errorCode]
```

Reply:
So bassRequest === 'DONE' is populated at the point a user selects either

Reply:
So `bassRequest === 'DONE'` is populated at the point a user selects either provided an address, CAS2 or opted out. As soon as one is selected they are all marked `DONE`.

![Screenshot 2024-12-13 at 12 57 22](https://github.com/user-attachments/assets/8ebcfd8d-5b69-46b1-badf-ffc017f1f47c)

Reply:
But we'll look into your suggestion above more 👀

Reply:
Think we need to do a bit of testing on it 🧪

Reply:
Updated block logic following above suggestion 😄